sunpy/data/test/__init__.py

```diff
@@ -60,3 +64,48 @@ def test_data_filenames():
         test_data_filenames_list.extend(files)
 
     return test_data_filenames_list
+
+
+def get_dummy_map_from_header(filename):
+    """
+    Generate a dummy `~sunpy.map.Map` from header-only test data.
+
+    The "image" will be random numbers with the correct shape
+    as specified by the header.
+    """
+    filepath = get_test_filepath(filename)
+    with open(filepath, 'r') as f:
+        header = astropy.io.fits.Header.fromstring(f.read())
+    data = np.random.rand(header['naxis1'], header['naxis2'])
+    # NOTE: by reading straight from the data header pair, we are skipping
+    # the fixes that are applied in sunpy.io.fits, e.g. the waveunit fix
+    # Would it be better to write this whole map back to a temporary file and then
+    # read it back in by passing in the filename instead?
+    return sunpy.map.Map(data, header)
+
+
+def convert_image_to_header_only(image_path, header_path, **kwargs):
+    """
+    Convert a FITS file containing an image and a header
+    to a plaintext file containing only the string representation
+    of the header.
+
+    This is to be used in the context of generating header-only test data
+    files.
+
+    Parameters
+    -----------
+    image_path : path-like
+        Path to original image data FITS file
+    header_path : path-like
+        Path to header-only test data plaintext file
+
+    See Also
+    --------
+    get_dummy_map_from_header
+    """
+    hdu = kwargs.get('hdu', 0)
+    with astropy.io.fits.open(image_path) as hdul:
+        header = hdul[hdu].header
+    with open(header_path, 'w') as f:
+        f.write(header.tostring())
```

Comment:
Would it be possible to write this with newlines between each key? I've found it helpful recently to be able to quickly scan FITS headers in the tests to see if they have e.g. any observer information, or if that info is in non-standard keys, which is much easier to do when the header isn't a single line of text.

Reply:
I've written them out in a much more human-readable format